lib/reactor/executor/hooks.ex

```diff
@@ -94,4 +103,11 @@ defmodule Reactor.Executor.Hooks do
 
     :ok
   end
+
+  defp step_count(reactor) when is_nil(reactor.plan), do: length(reactor.steps)
+
+  defp step_count(reactor) do
+    vertices = Graph.num_vertices(reactor.plan)
+    length(reactor.steps) + vertices
```

Comment:
Not sure why, but this seems strange to me. The number of steps is the vertices in the plan plus the steps?

Reply:
That's correct. Any steps that have already been planned will be in the graph, whereas any unplanned steps are stored in the `steps` list until the next loop through the reactor.

Reply:
Should we store a step count in the state as steps are added so that we don't have to do lots of counting? I'm not sure if `num_vertices/1` is constant time or not, but if not we might want to consider keeping both counts in the state. Probably not big lists, but also if we count every time for every invocation of every middleware, it could end up being lots of calls.

Reply:
pretty sure it just calls `map_size` behind the scenes. we _could_ cache it, but since `step_count` is only called by the `init` callback, it's not happening very often. 🤷